source/docs/prismic/embed-fields.md

```diff
@@ -0,0 +1,215 @@
+---
+id: embed-fields
+title: Embed Fields
+---
+
+Embed fields allows you to add a valid oEmbed URL, like YouTube, Vimeo, or Spotify, to generate embed html content, or add your own custom embed content. You can read more about the oEmbed format, view a list of supported providers on the [oEmbed website](https://oembed.com/).
+
+In Front-Commerce we expose the embed fields in two ways;
+
+- Standalone Embed fields with an EmbedTransformer
+- Embed Fields in [`WysiwygV2`](/docs/advanced/theme/wysiwyg.html#lt-WysiwygV2-gt-usage) with [`PrismicWysiwyg`](/docs/advanced/theme/wysiwyg-platform.html#PrismicWysiwyg)
+
+In this section we will cover how to implement both of these methods, and how to implement your own custom embed fields.
+
+## Standalone Embed Fields
+
+You can learn how to [configure an embed field](https://prismic.io/docs/core-concepts/embed) in the prismic documentation.
+
+For this example let's say we want to create an album cover from an embed field.
+
+### Adding an Embed Field in your server-side
+
+First add an embed field in your schema using the [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type, for example:
+
+```graphql
+type MyAlbum {
+  title: String
+  artist: String
+  releaseDate: Date
+  cover: oEmbedContent
+}
+```
+
+Then you can implement the [`EmbedTransformer`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/loaders/transformers/Embed.js) in your loader which will parse the embed response from Prismic to a rich [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type.
+
+```js
+const AlbumLoader = (PrismicLoader) => {
+  const { EmbedTransformer } = PrismicLoader.transformers;
+
+  const contentTransformOptions = {
+    fieldTransformers: {
+      // ... other transformers
+      cover: new EmbedTransformer(),
+    },
+  };
+
+  return {
+    loadByUID: (uid) => {
+      return PrismicLoader.loadByUID("album", uid, contentTransformOptions);
+    },
+  };
+};
+```
+
+### Adding an Embed Field in your client-side
+
+We have added a `PrismicEmbed` component to handle the embed response in you client side.
+
+To simplify the usage between the component and the backend data we created a `PrismicEmbedFragment` which contains the minimum required data for the component, for example:
+
+```graphql
+#import "theme/modules/prismic/PrismicEmbed/PrismicEmbedFragment"
+
+fragment AlbumFragment on Album {
+  title
+  cover {
+    ...PrismicEmbedFragment
+  }
+}
+```
+
+The fragment will expose the `html`, `provider` and `type` fields in the [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type.
```

Comment:
**issue(non-blocking):** this list might become outdated (out of sync from the code) over time.

**suggestion:** rephrase it in a more vague way
```suggestion
The fragment will expose several fields from the [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type.
```